src/geode/model/helpers/convert_to_mesh.cpp

```diff
@@ -539,6 +553,319 @@ namespace
+                model2mesh.unique_vertices_mapping.in2out( corner_uv );
+            model2mesh.corner_vertices_mapping.map(
+                { corner.id(), 0 }, solid_uv );
+        }
+    }
+
+    template < typename Model >
+    void build_edges_from_model( const Model& model,
+        geode::ModelToMeshMappings& model2mesh,
+        geode::EdgedCurveBuilder< Model::dim >& mesh_builder )
+    {
+        for( const auto& line : model.lines() )
+        {
+            const auto& line_mesh = line.mesh();
+            for( const auto& e : geode::Range{ line_mesh.nb_edges() } )
+            {
+                std::array< geode::index_t, 2 > vertices;
+                for( const auto v : geode::LRange{ 2 } )
+                {
+                    const auto vertex = line_mesh.edge_vertex( { e, v } );
+                    const auto unique_vertex =
+                        model.unique_vertex( { line.component_id(), vertex } );
+                    if( model2mesh.unique_vertices_mapping.has_mapping_input(
+                            unique_vertex ) )
+                    {
+                        vertices[v] = model2mesh.unique_vertices_mapping.in2out(
+                            unique_vertex );
+                    }
+                    else
+                    {
+                        const auto new_vertex_index = mesh_builder.create_point(
+                            line_mesh.point( vertex ) );
+                        vertices[v] = new_vertex_index;
+                        model2mesh.unique_vertices_mapping.map(
+                            unique_vertex, new_vertex_index );
+                    }
+                }
+                const auto edge_index =
+                    mesh_builder.create_edge( vertices[0], vertices[1] );
+                model2mesh.line_edges_mapping.map(
+                    { line.id(), e }, edge_index );
+            }
+        }
+    }
+
+    template < typename Model >
+    std::tuple< std::unique_ptr< geode::EdgedCurve< Model::dim > >,
+        geode::ModelToMeshMappings >
+        new_convert_model_into_curve( const Model& model )
+    {
+        geode::ModelToMeshMappings model2mesh;
+        std::vector<
+            std::reference_wrapper< const geode::EdgedCurve< Model::dim > > >
+            meshes;
+        meshes.reserve( model.nb_lines() );
+        for( const auto& line : model.lines() )
+        {
+            meshes.emplace_back( line.mesh() );
+        }
+        auto mesh =
+            geode::detail::create_mesh< geode::EdgedCurve< Model::dim > >(
+                std::move( meshes ) );
+        auto mesh_builder =
+            geode::EdgedCurveBuilder< Model::dim >::create( *mesh );
+        build_edges_from_model( model, model2mesh, *mesh_builder );
+        map_corner_vertices( model, model2mesh );
+        return std::make_pair( std::move( mesh ), std::move( model2mesh ) );
+    }
+
+    template < geode::index_t dim >
+    void compute_polygons_surface_adjacencies(
```

Comment:
There is no computation here, maybe rename function to something like "set_..." or "reuse_..."